lib/nrf_modem_lib/lte_connectivity/Kconfig

```diff
@@ -33,6 +33,17 @@ config LTE_CONNECTIVITY_CONNECT_TIMEOUT_SECONDS
 	  When the timeout is disabled, the library relies solely on the modem's internal
 	  connection handling to upkeep the connection.
 
+config LTE_CONNECTIVITY_REGISTRATION_TIMEOUT_SECONDS
```

Comment:
Given CEREG 2 is ignored, I'll provoke a discussion whether we need a timeout at all? :)

Reply:
That's a good question. Modem will send `+CEREG: 4` after the search has been finished and nothing was found. The next search starts depending on the periodic search configuration. With the default configuration, I think the delay after the first search is only 10 seconds and starts increasing from that. With this configuration, it would be possible to wait for several searches to be done before determining that LTE connectivity has been lost. But I'm not sure if that's necessary. With `+CEREG: 2` ignored, I guess the timeout could be removed and the implementation simplified.

Reply:
Oooh, I'm of two minds on this. For my purposes a long timeout is not needed, and removing the timeout would be nice.

But as @tokangas points out, I could definitely see a customer wanting to maybe wait for a few searches before officially timing out.

Plus, the timeout setting is a convenient way for the customer to disable this behavior altogether, for instance if they want to rely solely on PDN state.

Does any of this change your mind @trantanen? If not then I can remove the timeout and we can always throw it back in if customers ask for it.

This would have the side effect of making CEREG tracking mandatory, though -- no way to pay attention only to PDN -- which admittedly seems way less likely to be that interesting for a customer IMO

Reply:
> Does any of this change your mind @trantanen? 
> Oooh, I'm of two minds on this.

Same here :) I was just initiating the conversation as I wasn't sure either whether we would need it but...

> If not then I can remove the timeout and we can always throw it back in if customers ask for it.

...unless we come up with a very valid use case, this is the reason why I would leave it out for now. If we add it now, there's a high chance it's one of those dozens of Kconfig's that Zephyr+NCS has that nobody ever use.

> This would have the side effect of making CEREG tracking mandatory, though admittedly disabling it seems like an unlikely desire

This is what I've been thinking, i.e., what's the drawback for the client if net_if is set to dormant state when network is not available? I'm not net_if guru so I'm curious.
And again, if this becomes a problem, we can add a Kconfig.

Reply:
In fact, thinking on it, disabling CEREG monitoring basically renders `lte_connectivity` unable to ***ever*** detect network loss if the cell network is totally unreachable, so I'm not sure that **_any_** customers would ever find it valuable. They may as well just not use conn_mgr at that point.

Let's ditch the timeout then and fully adopt CEREG monitoring, you've convinced me

Reply:
Resolving since it is implemented